### ISSUE TITLE
SSL error when trying to send email using SMTP

### ISSUE THREAD

**Issue by codejamninja** [tomav/docker-mailserver#1566]
I get the following server error when trying to send an email using SMTP.

```
334F9277: SSL error:0D06B08E:asn1 encoding routines:asn1_d2i_read_bio:not enough data
334F9277: dkim_eom(): resource unavailable: d2i_PrivateKey_bio() failed
```

When I try to send an email using telnet `openssl s_client` the connection breaks immediate after adding `RCPT TO`.

```
RCPT TO: email@example.com
RENEGOTIATING
140649841022784:error:1420410A:SSL routines:SSL_renegotiate:wrong ssl version:ssl/ssl_lib.c:2127:
```

Please note that I am able to login to SMTP successfully.

Full client log . . .
```
openssl s_client -starttls smtp -crlf -connect mail.siliconhills.co:5587

EHLO localhost
250-mail.siliconhills.co
250-PIPELINING
250-SIZE 10240000
250-ETRN
250-AUTH PLAIN LOGIN
250-AUTH=PLAIN LOGIN
250-ENHANCEDSTATUSCODES
250-8BITMIME
250-DSN
250 CHUNKING
AUTH LOGIN
334 VXNlcm5hbWU6
amFtQHNpbGljb25oaWxscy5jbw==
334 UGFzc3dvcmQ6
cGFzcw==
235 2.7.0 Authentication successful
MAIL FROM: jam@siliconhills.co
250 2.1.0 Ok
RCPT TO: email@example.com
RENEGOTIATING
140649841022784:error:1420410A:SSL routines:SSL_renegotiate:wrong ssl version:ssl/ssl_lib.c:2127:
```

Full log server . . .
```
Jul 12 21:38:56 mailserver7-mailserver-mailserver-7596b45984-7pql7 dovecot: imap-login: Login: user=<jam@siliconhills.co>, method=PLAIN, rip=165.227.51.171, lip=10.42.4.78, mpid=4608, TLS, session=<fri0ZkWqaIil4zOr>
Jul 12 21:38:56 mailserver7-mailserver-mailserver-7596b45984-7pql7 postfix/smtps/smtpd[4513]: connect from 165-227-51-171.expose-kubelets-metrics.cattle-prometheus.svc.cluster.local[165.227.51.171]
Jul 12 21:38:56 mailserver7-mailserver-mailserver-7596b45984-7pql7 postfix/smtps/smtpd[4513]: Anonymous TLS connection established from 165-227-51-171.expose-kubelets-metrics.cattle-prometheus.svc.cluster.local[165.227.51.171]: TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits)
Jul 12 21:38:56 mailserver7-mailserver-mailserver-7596b45984-7pql7 postfix/smtps/smtpd[4513]: 334F9277: client=165-227-51-171.expose-kubelets-metrics.cattle-prometheus.svc.cluster.local[165.227.51.171], sasl_method=PLAIN, sasl_username=jam@siliconhills.co
Jul 12 21:38:56 mailserver7-mailserver-mailserver-7596b45984-7pql7 postfix/sender-cleanup/cleanup[4514]: 334F9277: replace: header MIME-Version: 1.0 from 165-227-51-171.expose-kubelets-metrics.cattle-prometheus.svc.cluster.local[165.227.51.171]; from=<jam@siliconhills.co> to=<email@example.com> proto=ESMTP helo=<mail.siliconhills.co>: Mime-Version: 1.0
Jul 12 21:38:56 mailserver7-mailserver-mailserver-7596b45984-7pql7 postfix/sender-cleanup/cleanup[4514]: 334F9277: message-id=<7d422cf428cce6928e79d0f1cc6e88ae@siliconhills.co>
Jul 12 21:38:56 mailserver7-mailserver-mailserver-7596b45984-7pql7 opendkim[230]: 334F9277: SSL error:0D06B08E:asn1 encoding routines:asn1_d2i_read_bio:not enough data
Jul 12 21:38:56 mailserver7-mailserver-mailserver-7596b45984-7pql7 opendkim[230]: 334F9277: dkim_eom(): resource unavailable: d2i_PrivateKey_bio() failed
Jul 12 21:38:56 mailserver7-mailserver-mailserver-7596b45984-7pql7 postfix/sender-cleanup/cleanup[4514]: 334F9277: milter-reject: END-OF-MESSAGE from 165-227-51-171.expose-kubelets-metrics.cattle-prometheus.svc.cluster.local[165.227.51.171]: 4.7.0 resource unavailable; from=<jam@siliconhills.co> to=<email@example.com> proto=ESMTP helo=<mail.siliconhills.co>
```

**Comment by erik-wramner** [comment]
Looking at the logs the error is actually from opendkim. Have you configured your opendkim keys correctly, see [ReadMe](https://github.com/tomav/docker-mailserver)?

**Comment by codejamninja** [comment]
Yes, I tried to setup opendkim, but I am having the following issue.

https://github.com/tomav/docker-mailserver/issues/1565

**Comment by erik-wramner** [comment]
Right. Then this problem will hopefully go away when the other issue is solved.

**Comment by codejamninja** [comment]
@erik-wramner do you have any insights on the other issue?

**Comment by erik-wramner** [comment]
Not on top of my head. Unfortunately I don't have much time, but I would suggest trying to reproduce the issue from a clean slate. If it is reproducible we should be able to find the problem and fix it. Most likely the script fails on something before it can write the keys.

**Comment by aendeavor** [comment]
@codejamninja A few changes were done, but not specifically to this issue. But seeing #1565 has a possible fix, could you retry with the current master?

**Comment by codejamninja** [comment]
@aendeavor yes I'll try again in a few days. Thanks!

**Comment by aendeavor** [comment]
Any results here?

**Comment by aendeavor** [comment]
This issue was closed due to one or more of the following reasons:

1. Age
2. Contributor inactivity
3. The issue seems to be resolved

If you think this happened by accident, or feel like this issue was not actually resolved, please feel free to re-open it. If there is an issue you could resolve in the meantime, please open a PR based on the current master branch so we can review it.

**Comment by codejamninja** [comment]
@aendeavor I finally got around to testing again and am still have this same issue.

**Comment by aendeavor** [comment]
Alright. What's the version you're using now? This seems to me like a problem with either non-matching SSL/TSL version (as a result of misconfiguration - `TLS_LEVEL`?) or related to OpenDKIM in a strange way...

**Comment by codejamninja** [comment]
I figued out my issue. I was trying to generate a `ed25519` private key using the helm `genPrivateKey` function. Apparently my version of helm doesn't support `ed25519` private keys. I switched to `rsa` and it seems to work fine.

```
genPrivateKey "rsa"
```

https://masterminds.github.io/sprig/crypto.html#genprivatekey

https://github.com/codejamninja/charts/blob/ddf98befa8d40c8ac0cd97016a5d6726d56089da/beta/mailserver/templates/secrets/mailserver.yaml#L12
